cpp/src/replace/replace.cu

```diff
@@ -424,13 +475,12 @@ struct replace_nulls_scalar_kernel_forwarder {
 
     using ScalarType = cudf::experimental::scalar_type_t<col_type>;
     auto s1 = static_cast<ScalarType const&>(replacement);
-
-    replace_nulls<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(nrows,
-                                                              input.data<col_type>(),
-                                                              input.null_mask(),
-                                                              thrust::make_constant_iterator(s1.value()),
-                                                              output_view.data<col_type>());
-    cudaStreamSynchronize(stream);
+    device_constant_iterator<col_type> iter(s1.data());
+    replace_nulls_functor<col_type, device_constant_iterator<col_type>> func(output_view.data<col_type>(),
+                                                                             input.data<col_type>(),
+                                                                             input.null_mask(),
+                                                                             iter);
+    thrust::for_each_n(thrust::make_counting_iterator(0), input.size(), func);
```

Comment:
This would be a little more intuitive. 
```
auto replacement_view = get_scalar_device_view(s1);
thrust::transform(input.begin<col_type>(), input.end<col_type>(), make_validity_iterator(input), output_view.begin<col_type>(), [replacement_view] __device__ (col_type input_element, bool is_valid) { return (is_valid) ? input_element : replacement_view.value(); }); 
```

Reply:
Changed to use thrust::transform